### PR TITLE
Grouped fields

### PR DESCRIPTION
**Pull request by danrot**
Once finished this PR enables to use the data mapper correctly in combination with checkboxes and radio buttons.

__Work left__
- [x] return only values which are checked or selected in `ElementGroup`
- [x] implement/use `ElementGroup` in collections
- [ ] ~~refactor to split elements, elementGroups and collections in code~~ (still a good idea, created issue #54)
- [ ] remove `$el` filter from `getData`, if not need anymore
- [x] Check BC for existing demos
- [ ] implement a few tests (?)
- [x] Test append resp. insertAfterLast from addToCollection resp. appendChildren

### CODE REVIEW

**Comment by danrot** [comment]
@wachterjohannes Would it be a big problem to remove the `$el` and `returnMapperId` from the `getData` method of the form? I've rebuild the blocks in sulu, so that they don't make use of this anymore, are there any other cases where this would be important?

Maybe I should also add that the `$el` isn't doing anything anymore, and I have never understood what `returnMapperId` really does.

File: js/elementGroup.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            for (var i = -1; ++i < elements.length;) {
+                if (elements[i].$el.val() === value) {
+                    elements[i].$el.prop('checked', true);
+
+                    return;
+                }
+            }
+        };
+
+    return function(elements, singleValue) {
```
Comment (wachterjohannes) [inline js/elementGroup.js]: is this `singleValue` a boolean flag?

File: js/mapper.js [hunk 2]
```diff
@@ -374,9 +414,11 @@ define([
...
-                        if (this.elements[i].data('mapper-id') === mapperId) {
-                            return this.elements[i];
+                    for (var i = -1, iLength = this.collections.length; ++i < iLength;) {
+                        for (var j = -1, jLength = this.collections[i].items.length; ++j < jLength;) {
+                            if (this.collections[i].items[j].data('mapper-id') === mapperId) {
+                                return this.collections[i].items[j];
+                            }
                         }
                     }
                     return null;
```
Comment (wachterjohannes) [inline js/mapper.js]: add empty line before return

File: js/mapper.js [hunk 3]
```diff
@@ -494,12 +503,12 @@ define([
...
                             if ($.isArray(value) && this.templates.hasOwnProperty(key)) {
                                 collection = this.templates[key].collection;
-                                collection.$child = this.templates[key].tpl;
+                                collection.key = key;
 
                                 // if first element of collection, clear collection
                                 if (!this.collectionsSet.hasOwnProperty(collection.id)) {
                                     collection.$element.children().each(function(key, value) {
-                                        that.remove.call(this, $(value));
+                                        $(value).remove();Reso
```
Comment (wachterjohannes) [inline js/mapper.js]: i think this `Reso` should not be there :)

File: js/elementGroup.js (hunk 1 above)
Comment (danrot) [inline js/elementGroup.js]: Yes, should I maybe rename it to `isSingleValue`?

File: js/mapper.js (hunk 3 above)
Comment (danrot) [inline js/mapper.js]: Strange that it still worked :confused:

File: js/elementGroup.js (hunk 1 above)
Comment (wachterjohannes) [inline js/elementGroup.js]: yes that would be more explicitly.

**Comment by danrot** [comment]
@wachterjohannes done

**Pull request merged**